ambari-server/src/main/java/org/apache/ambari/server/security/ldap/AmbariLdapDataPopulator.java

```diff
@@ -447,10 +447,17 @@ protected void refreshGroupMembers(LdapBatchDto batchInfo, LdapGroupDto group, M
    * @return the set of LDAP groups for the given name
    */
   protected Set<LdapGroupDto> getLdapGroups(String groupName) {
+    String baseDn = ldapServerProperties.getBaseDN().toLowerCase();
```

Comment:
It seems like we have some confusing properties here...
* baseDN
* userBase
* groupBase

My guess is that `userBase` and `groupBase` should be used, but if null they should revert to `baseDN`.  However, maybe `baseDN` should just go away?

@smolnar82 , any opinion on this?

Reply:
Before we migrated LDAP related properties in the DB we only supported `authentication.ldap.baseDn` and then we introduced two new properties called `ambari.ldap.attributes.group.search_base` and `ambari.ldap.attributes.user.search_base` in 2.7; these properties are set to the value of `authentication.ldap.baseDn` upon upgrade and can be modified later on using the REST API.

We use `ambari.ldap.attributes.user.search_base` as baseDn in `org.apache.ambari.server.security.authorization.LdapServerProperties` which is used all over the places when ldapServerProperties.getBaseDN() is invoked. Fortunately the method names where this invocation happens are quite helpful so I believe we can get rid of the `baseDN` approach and use either `ambari.ldap.attributes.group.search_base` or `ambari.ldap.attributes.user.search_base`.

In `org.apache.ambari.server.security.authorization.AmbariLdapAuthenticationProvider.loadLdapAuthenticationProvider(String)` we use 'baseDn' like this:
`springSecurityContextSource.setBase(ldapServerProperties.get().getBaseDN());`

Per the Spring JavaDoc:
>> Set the base suffix from which all operations should origin. If a base suffix is set, you will not have to (and, indeed, must not) specify the full distinguished names in any operations performed.

As far as I understood this means that if we do not set it then we should use the FDN's for any operation and this is exactly what we are doing if we will use the `ambari.ldap.attributes.group.search_base` or `ambari.ldap.attributes.user.search_base` as described above. Any objection?

We also have two other properties:
1. `ambari.ldap.attributes.search_user_base` (set to the value of `authentication.ldap.userBase` from ambari.properties upon upgrade and can be modified later on using the REST API)
1. `ambari.ldap.attributes.search_group_base` (set to the value of `authentication.ldap.groupBase` from ambari.properties upon upgrade and can be modified later on using the REST API)

Currently only the groupBase property is in use (in `org.apache.ambari.server.security.authorization.AmbariLdapBindAuthenticator.setAmbariAdminAttr(DirContextOperations, LdapServerProperties))`